### PR TITLE
fix: bug fixes related to zap scanning startup

### PR DESCRIPTION
**Pull request by mohamed-cds**
Zap scan lambda will read from the SNS topic and launch a new ECS task that will scan the URL provided and store the results in the owasp-zap report bucket. Reading the report from the bucket and storing it in the database and optionally aws security hub will be done seperately.

- Install glibc in alpine container so that awscliv2 installs correctly
- Fixes for S3 iam permissions
- Fixed container definition to reference correct docker image

### CODE REVIEW

File: runners/owasp-zap/Dockerfile [hunk 1]
```diff
@@ -2,18 +2,36 @@ FROM python:3.8-alpine
...
 WORKDIR /scan
 
+ENV GLIBC_VER=2.31-r0
+ARG AWS_CLI_VERSION="2.2.29"
+
+# install glibc compatibility for alpine (Needed for awscliv2)
 RUN apk update \
     && apk upgrade \
-    && apk add --update curl bash jq
+    && apk add --update curl binutils bash jq \
```
Comment (patheard) [inline runners/owasp-zap/Dockerfile:11]: You should be able to switch the `--update` to `--no-cache` which does an implicit `apk update` first and doesn't use a local index.

File: runners/owasp-zap/Dockerfile [hunk 2]
```diff
@@ -2,18 +2,36 @@ FROM python:3.8-alpine
...
+        awscliv2.zip \
+        aws \
+        /usr/local/aws-cli/v2/*/dist/aws_completer \
+        /usr/local/aws-cli/v2/*/dist/awscli/data/ac.index \
+        /usr/local/aws-cli/v2/*/dist/awscli/examples \
+    && apk --no-cache del \
+        binutils \
+    && rm glibc-${GLIBC_VER}.apk \
+    && rm glibc-bin-${GLIBC_VER}.apk \
+    && rm -rf /var/cache/apk/*
```
Comment (patheard) [inline runners/owasp-zap/Dockerfile:31]: Once all the `apk add` commands are using `--no-cache`, you can get rid of this line.

File: runners/owasp-zap/Dockerfile [hunk 3]
```diff
@@ -2,18 +2,36 @@ FROM python:3.8-alpine
 
 WORKDIR /scan
 
+ENV GLIBC_VER=2.31-r0
```
Comment (patheard) [inline runners/owasp-zap/Dockerfile:5]: If you only need `GLIBC_VER` when building the image, you can switch this to an ARG:
```dockerfile
ARG GLIBC_VER="2.31-r0"
```

**Review by patheard — COMMENTED** [review]
Looks good!  Had a few comments for the `Dockerfile` but nothing major.

File: runners/owasp-zap/Dockerfile (hunk 3 above)
Comment (mohamed-cds) [reply runners/owasp-zap/Dockerfile]: changed, thanks!

File: runners/owasp-zap/Dockerfile (hunk 2 above)
Comment (mohamed-cds) [reply runners/owasp-zap/Dockerfile]: changed, thanks!

File: runners/owasp-zap/Dockerfile (hunk 1 above)
Comment (mohamed-cds) [reply runners/owasp-zap/Dockerfile]: changed, thanks!

**Review by patheard — APPROVED** [review]

File: runners/owasp-zap/Dockerfile [hunk 4]
```diff
@@ -2,19 +2,33 @@ FROM python:3.8-alpine
...
-ARG AWS_CLI_VERSION="2.2.29"
-RUN curl -Lo awscliv2.zip "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${AWS_CLI_VERSION}.zip" \
-    && unzip awscliv2.zip \
-    && ./aws/install -i /usr/local/aws-cli -b /usr/local/bin
-
 COPY entrypoint.sh /entrypoint.sh
 
 # Launch OWASP scan
 ENTRYPOINT ["/entrypoint.sh"]
+# ENTRYPOINT ["tail", "-f", "/dev/null"]
```
Comment (dj2) [inline runners/owasp-zap/Dockerfile:34]: Remove?

**Review by dj2 — APPROVED** [review]
lgtm w/ nit

File: runners/owasp-zap/Dockerfile (hunk 4 above)
Comment (mohamed-cds) [reply runners/owasp-zap/Dockerfile]: removed

**Pull request merged**